
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.19: https://docutils.sourceforge.io/" />

    <title>BMtools &#8212; BMtools 0.1 documentation</title>
    <link rel="stylesheet" type="text/css" href="../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../_static/basic_mod.css?v=0.7.0-1" />
    <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
    <script src="../_static/jquery.js"></script>
    <script src="../_static/underscore.js"></script>
    <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
    <script src="../_static/doctools.js"></script>
    <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <script src="../_static/js/petite-vue.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Bam2bm" href="bam2bm.html" />
    <link rel="prev" title="The Basic DNA Methylation (BM) format and BMtools" href="../index.html" /> 
  </head><body data-dark_mode_code_blocks="true">

<div id="top_nav">
    

    <nav>
        
            
        

        <p id="toggle_sidebar">
            <a href="#" title="Toggle sidebar">|||</a>
        </p>
        <h1><a href="../index.html" title="Go to homepage">BMtools 0.1 documentation</a></h1>

        <a id="mode_toggle" href="#" @click.prevent="handleClick" :title="mode">
    <template v-if="mode == 'light'">
        <svg width="100%" height="100%" viewBox="0 0 79 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><g id="mode_light"><rect id="Bounds" x="0" y="-0" width="78.623" height="79.049" style="fill:none;"/><circle cx="39.311" cy="39.524" r="15.734" style="fill:#fff;"/><g id="beams"><g id="beam"><path id="beam1" serif:id="beam" d="M44.212,4.901c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.9,2.196 -4.9,4.901l-0,9.614c-0,2.705 2.196,4.901 4.9,4.901c2.705,0 4.901,-2.196 4.901,-4.901l0,-9.614Z" style="fill:#fff;"/></g><g id="beam2" serif:id="beam"><path id="beam3" serif:id="beam" d="M67.48,18.073c1.913,-1.912 1.913,-5.018 0,-6.931c-1.912,-1.912 -5.018,-1.912 -6.931,0l-6.798,6.799c-1.912,1.912 -1.912,5.018 0,6.931c1.913,1.912 5.018,1.912 6.931,-0l6.798,-6.799Z" style="fill:#fff;"/></g><g id="beam4" serif:id="beam"><path id="beam5" serif:id="beam" d="M25.728,61.108c1.912,-1.913 1.912,-5.018 -0,-6.931c-1.913,-1.913 -5.019,-1.913 -6.931,-0l-6.799,6.798c-1.912,1.913 -1.912,5.019 0,6.931c1.913,1.913 5.019,1.913 6.931,0l6.799,-6.798Z" style="fill:#fff;"/></g><g id="beam6" serif:id="beam"><path id="beam7" serif:id="beam" d="M60.682,54.177c-1.913,-1.913 -5.018,-1.913 -6.931,-0c-1.912,1.913 -1.912,5.018 0,6.931l6.798,6.798c1.913,1.913 5.019,1.913 6.931,0c1.913,-1.912 1.913,-5.018 0,-6.931l-6.798,-6.798Z" style="fill:#fff;"/></g><g id="beam8" serif:id="beam"><path id="beam9" serif:id="beam" d="M4.901,34.623c-2.705,0 -4.901,2.196 -4.901,4.901c0,2.705 2.196,4.901 4.901,4.901l9.614,0c2.705,0 4.901,-2.196 4.901,-4.901c0,-2.705 -2.196,-4.901 -4.901,-4.901l-9.614,0Z" style="fill:#fff;"/></g><g id="beam10" serif:id="beam"><path id="beam11" serif:id="beam" d="M44.212,64.534c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.9,2.196 -4.9,4.901l-0,9.614c-0,2.705 2.196,4.901 4.9,4.901c2.705,-0 4.901,-2.196 4.901,-4.901l0,-9.614Z" style="fill:#fff;"/></g><g id="beam12" serif:id="beam"><path id="beam13" serif:id="beam" d="M18.929,11.142c-1.912,-1.912 -5.018,-1.912 -6.931,0c-1.912,1.913 -1.912,5.019 0,6.931l6.799,6.799c1.912,1.912 5.018,1.912 6.931,-0c1.912,-1.913 1.912,-5.019 -0,-6.931l-6.799,-6.799Z" style="fill:#fff;"/></g><g id="beam14" serif:id="beam"><path id="beam15" serif:id="beam" d="M64.108,34.623c-2.705,0 -4.901,2.196 -4.901,4.901c-0,2.705 2.196,4.901 4.901,4.901l9.614,0c2.705,0 4.901,-2.196 4.901,-4.901c-0,-2.705 -2.196,-4.901 -4.901,-4.901l-9.614,0Z" style="fill:#fff;"/></g></g></g></svg>
    </template>

    <template v-if="mode == 'dark'">
        <svg width="100%" height="100%" viewBox="0 0 79 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><g id="mode_dark"><rect id="Bounds" x="0" y="-0" width="78.623" height="79.049" style="fill:none;"/><circle cx="39.311" cy="39.524" r="15.734" style="fill:#fff;"/><g id="beams"><g id="beam"><path id="beam1" serif:id="beam" d="M44.212,14.515c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,0 -4.901,2.196 -4.901,4.901c0,2.705 2.197,4.901 4.901,4.901c2.705,0 4.901,-2.196 4.901,-4.901Z" style="fill:#fff;"/></g><g id="beam2" serif:id="beam"><path id="beam3" serif:id="beam" d="M60.662,24.892c1.902,-1.902 1.902,-4.99 0,-6.892l-0.04,-0.039c-1.901,-1.902 -4.989,-1.902 -6.891,-0c-1.901,1.901 -1.901,4.989 0,6.891l0.04,0.04c1.902,1.901 4.989,1.901 6.891,-0Z" style="fill:#fff;"/></g><g id="beam4" serif:id="beam"><path id="beam5" serif:id="beam" d="M25.732,61.103c1.91,-1.91 1.91,-5.011 0,-6.921l-0.009,-0.01c-1.91,-1.91 -5.012,-1.91 -6.921,-0c-1.91,1.91 -1.91,5.011 -0,6.921l0.01,0.01c1.909,1.91 5.011,1.91 6.92,-0Z" style="fill:#fff;"/></g><g id="beam6" serif:id="beam"><path id="beam7" serif:id="beam" d="M60.672,54.167c-1.907,-1.907 -5.004,-1.907 -6.911,0l-0.02,0.02c-1.907,1.907 -1.907,5.004 0,6.911c1.907,1.907 5.004,1.907 6.911,-0l0.02,-0.02c1.907,-1.907 1.907,-5.004 0,-6.911Z" style="fill:#fff;"/></g><g id="beam8" serif:id="beam"><path id="beam9" serif:id="beam" d="M14.52,34.623c-2.702,0 -4.896,2.194 -4.896,4.896l0,0.01c0,2.702 2.194,4.896 4.896,4.896c2.702,0 4.896,-2.194 4.896,-4.896l-0,-0.01c-0,-2.702 -2.194,-4.896 -4.896,-4.896Z" style="fill:#fff;"/></g><g id="beam10" serif:id="beam"><path id="beam11" serif:id="beam" d="M44.212,64.534c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.901,2.196 -4.901,4.901c0,2.704 2.197,4.9 4.901,4.9c2.705,0 4.901,-2.196 4.901,-4.9Z" style="fill:#fff;"/></g><g id="beam12" serif:id="beam"><path id="beam13" serif:id="beam" d="M25.73,17.943c-1.911,-1.911 -5.015,-1.911 -6.926,0l-0.005,0.005c-1.911,1.911 -1.911,5.015 0,6.926c1.911,1.911 5.015,1.911 6.926,0l0.005,-0.005c1.911,-1.911 1.911,-5.014 -0,-6.926Z" style="fill:#fff;"/></g><g id="beam14" serif:id="beam"><path id="beam15" serif:id="beam" d="M64.098,34.623c-2.699,0 -4.891,2.192 -4.891,4.892l-0,0.019c-0,2.699 2.192,4.891 4.891,4.891c2.7,0 4.892,-2.192 4.892,-4.891l0,-0.019c0,-2.7 -2.192,-4.892 -4.892,-4.892Z" style="fill:#fff;"/></g></g></g></svg>
    </template>

    <template v-if="mode == 'darkest'">
        <svg width="100%" height="100%" viewBox="0 0 79 80" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve" style="fill-rule:evenodd;clip-rule:evenodd;stroke-linejoin:round;stroke-miterlimit:2;"><g id="mode_darkest"><rect id="Bounds" x="0" y="-0" width="78.623" height="79.049" style="fill:none;"/><path d="M39.315,23.791c8.684,-0 15.734,7.05 15.734,15.733c0,8.684 -7.05,15.734 -15.734,15.734c-8.683,0 -15.733,-7.05 -15.733,-15.734c-0,-8.683 7.05,-15.733 15.733,-15.733Zm0,4.737c6.069,0 10.997,4.927 10.997,10.996c-0,6.069 -4.928,10.996 -10.997,10.996c-6.068,0 -10.996,-4.927 -10.996,-10.996c0,-6.069 4.928,-10.996 10.996,-10.996Z" style="fill:#fff;"/><g id="beams"><g id="beam"><path id="beam1" serif:id="beam" d="M44.216,14.515c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,0 -4.9,2.196 -4.9,4.901c-0,2.705 2.196,4.901 4.9,4.901c2.705,0 4.901,-2.196 4.901,-4.901Z" style="fill:#fff;"/></g><g id="beam2" serif:id="beam"><path id="beam3" serif:id="beam" d="M60.666,24.892c1.902,-1.902 1.902,-4.99 0,-6.892l-0.04,-0.039c-1.901,-1.902 -4.989,-1.902 -6.891,-0c-1.901,1.901 -1.901,4.989 0,6.891l0.04,0.04c1.902,1.901 4.99,1.901 6.891,-0Z" style="fill:#fff;"/></g><g id="beam4" serif:id="beam"><path id="beam5" serif:id="beam" d="M25.737,61.103c1.909,-1.91 1.909,-5.011 -0,-6.921l-0.01,-0.01c-1.91,-1.91 -5.011,-1.91 -6.921,-0c-1.91,1.91 -1.91,5.011 -0,6.921l0.01,0.01c1.91,1.91 5.011,1.91 6.921,-0Z" style="fill:#fff;"/></g><g id="beam6" serif:id="beam"><path id="beam7" serif:id="beam" d="M60.676,54.167c-1.907,-1.907 -5.004,-1.907 -6.911,0l-0.02,0.02c-1.907,1.907 -1.907,5.004 0,6.911c1.907,1.907 5.004,1.907 6.911,-0l0.02,-0.02c1.907,-1.907 1.907,-5.004 0,-6.911Z" style="fill:#fff;"/></g><g id="beam8" serif:id="beam"><path id="beam9" serif:id="beam" d="M14.524,34.623c-2.702,0 -4.896,2.194 -4.896,4.896l0,0.01c0,2.702 2.194,4.896 4.896,4.896c2.702,0 4.896,-2.194 4.896,-4.896l0,-0.01c0,-2.702 -2.194,-4.896 -4.896,-4.896Z" style="fill:#fff;"/></g><g id="beam10" serif:id="beam"><path id="beam11" serif:id="beam" d="M44.216,64.534c0,-2.705 -2.196,-4.901 -4.901,-4.901c-2.704,-0 -4.9,2.196 -4.9,4.901c-0,2.704 2.196,4.9 4.9,4.9c2.705,0 4.901,-2.196 4.901,-4.9Z" style="fill:#fff;"/></g><g id="beam12" serif:id="beam"><path id="beam13" serif:id="beam" d="M25.734,17.943c-1.911,-1.911 -5.015,-1.911 -6.926,0l-0.005,0.005c-1.911,1.911 -1.911,5.015 0,6.926c1.911,1.911 5.015,1.911 6.926,0l0.005,-0.005c1.911,-1.911 1.911,-5.014 0,-6.926Z" style="fill:#fff;"/></g><g id="beam14" serif:id="beam"><path id="beam15" serif:id="beam" d="M64.103,34.623c-2.7,0 -4.892,2.192 -4.892,4.892l-0,0.019c-0,2.699 2.192,4.891 4.892,4.891c2.699,0 4.891,-2.192 4.891,-4.891l0,-0.019c0,-2.7 -2.192,-4.892 -4.891,-4.892Z" style="fill:#fff;"/></g></g></g></svg>
    </template>
</a>

<script>
(function() {
    const LOCAL_STORAGE_KEY = 'piccoloThemeMode'

    var initialMode = localStorage.getItem(LOCAL_STORAGE_KEY)

    if (initialMode) {
        // Make sure the value in local storage is valid
        if (['light', 'dark', 'darkest'].indexOf(initialMode) == -1) {
            initialMode = 'light'
            localStorage.setItem(LOCAL_STORAGE_KEY, initialMode)
        }
    } else {
        // Check if the client prefers dark mode
        if (window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches) {
            initialMode = 'dark'
        } else {
            initialMode = 'light'
        }
        localStorage.setItem(LOCAL_STORAGE_KEY, initialMode)
    }

    document.documentElement.dataset.mode = initialMode

    PetiteVue.createApp({
        'mode': initialMode,
        handleClick() {
            let currentMode = this.mode

            if (currentMode == 'light') {
                this.mode = 'dark'
            } else if (currentMode == 'dark') {
                this.mode = 'darkest'
            } else if (currentMode == 'darkest') {
                this.mode = 'light'
            }

            document.documentElement.dataset.mode = this.mode
            localStorage.setItem(LOCAL_STORAGE_KEY, this.mode)

            console.log(this.mode)
        }
    }).mount('#mode_toggle')
})()
</script>
            <p class="mobile_search_link">
                <a href="../search.html" title="Search">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 65 64" fill-rule="evenodd" stroke-linejoin="round" stroke-miterlimit="2">
                        <path d="M14.873 40.009c-2.315-3.943-3.642-8.532-3.642-13.429C11.231 11.91 23.141 0 37.811 0s26.58 11.91 26.58 26.58-11.91 26.58-26.58 26.58a26.44 26.44 0 0 1-14.277-4.161L9.739 62.794a3.12 3.12 0 0 1-4.413 0L.913 58.382c-1.217-1.218-1.217-3.196 0-4.413l13.96-13.96zM37.811 8.054c10.225 0 18.526 8.301 18.526 18.526s-8.301 18.526-18.526 18.526-18.526-8.301-18.526-18.526S27.586 8.054 37.811 8.054z" fill="#fff" />
                    </svg>
                </a>
            </p>
        

        <div class="searchbox_wrapper">
            
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>document.getElementById('searchbox').style.display = "block"</script>
        </div>
    </nav>
</div>

    
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper"><ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">BMtools</a></li>
<li class="toctree-l1"><a class="reference internal" href="bam2bm.html">Bam2bm</a></li>
<li class="toctree-l1"><a class="reference internal" href="PlotMeth.html">PlotMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="DiffMeth.html">DiffMeth</a></li>
<li class="toctree-l1"><a class="reference internal" href="Requirements.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="pybmtools.html">pybmtools</a></li>
</ul>

        </div>
      </div>


    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="bmtools">
<h1>BMtools<a class="headerlink" href="#bmtools" title="Permalink to this heading">¶</a></h1>
<nav class="contents local" id="contents">
<ul class="simple">
<li><p><a class="reference internal" href="#bmtools-main-modules" id="id1">BMtools main modules</a></p></li>
<li><p><a class="reference internal" href="#main-functions" id="id2">Main functions</a></p></li>
<li><p><a class="reference internal" href="#bmtools-bam2bm" id="id3">bmtools bam2bm</a></p></li>
<li><p><a class="reference internal" href="#bmtools-mr2bm" id="id4">bmtools mr2bm</a></p></li>
<li><p><a class="reference internal" href="#bmtools-view" id="id5">bmtools view</a></p></li>
<li><p><a class="reference internal" href="#bmtools-viewheader" id="id6">bmtools viewheader</a></p></li>
<li><p><a class="reference internal" href="#bmtools-overlap" id="id7">bmtools overlap</a></p></li>
<li><p><a class="reference internal" href="#bmtools-regionstats" id="id8">bmtools regionstats</a></p></li>
<li><p><a class="reference internal" href="#bmtools-bodystats" id="id9">bmtools bodystats</a></p></li>
<li><p><a class="reference internal" href="#bmtools-profile" id="id10">bmtools profile</a></p></li>
<li><p><a class="reference internal" href="#bmtools-chromstats" id="id11">bmtools chromstats</a></p></li>
<li><p><a class="reference internal" href="#bmtools-addzm" id="id12">bmtools addzm</a></p></li>
<li><p><a class="reference internal" href="#bmtools-stats" id="id13">bmtools stats</a></p></li>
</ul>
</nav>
<section id="bmtools-main-modules">
<h2><a class="toc-backref" href="#id1" role="doc-backlink">BMtools main modules</a><a class="headerlink" href="#bmtools-main-modules" title="Permalink to this heading">¶</a></h2>
<p>You can view and process bm file with bmtools:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools view -i mutant.methratio.bm <span class="p">|</span> head
</pre></div>
</div>
<p>obtained text format methylation results</p>
</section>
<section id="main-functions">
<h2><a class="toc-backref" href="#id2" role="doc-backlink">Main functions</a><a class="headerlink" href="#main-functions" title="Permalink to this heading">¶</a></h2>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head" colspan="2"><p><strong>[ Main paramaters ]</strong></p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td></td>
<td><p>bmtools &lt;mode&gt; [opnions]</p></td>
</tr>
<tr class="row-odd"><td colspan="2"><p>Usage:</p></td>
</tr>
<tr class="row-even"><td><p>[mode]</p></td>
<td><p>bam2bm mr2bm view overlap regionstats bodystats profile chromstats</p></td>
</tr>
<tr class="row-odd"><td><p>bam2bm</p></td>
<td><p>calculate DNA methylation (BM format) with BAM file</p></td>
</tr>
<tr class="row-even"><td><p>mr2bm</p></td>
<td><p>convert txt meth file to bm format</p></td>
</tr>
<tr class="row-odd"><td><p>view</p></td>
<td><p>bm format to txt meth</p></td>
</tr>
<tr class="row-even"><td><p>viewheader</p></td>
<td><p>view header of bm file</p></td>
</tr>
<tr class="row-odd"><td><p>overlap</p></td>
<td><p>overlap cytosine site with more than two bm files</p></td>
</tr>
<tr class="row-even"><td><p>regionstats</p></td>
<td><p>calculate DNA methylation level of per region</p></td>
</tr>
<tr class="row-odd"><td><p>bodystats</p></td>
<td><p>calculate DNA methylation level of body, upstream and downstream.</p></td>
</tr>
<tr class="row-even"><td><p>profile</p></td>
<td><p>calculate DNA methylation profile</p></td>
</tr>
<tr class="row-odd"><td><p>chromstats</p></td>
<td><p>calculate DNA methylation level across chromosome</p></td>
</tr>
<tr class="row-even"><td><p>addzm</p></td>
<td><p>add or change zoom levels for bm format, need for browser visulization</p></td>
</tr>
<tr class="row-odd"><td><p>stats</p></td>
<td><p>coverage and methylation level distribution of data</p></td>
</tr>
<tr class="row-even"><td colspan="2"><p>-h|--help</p></td>
</tr>
</tbody>
</table>
</section>
<section id="bmtools-bam2bm">
<h2><a class="toc-backref" href="#id3" role="doc-backlink">bmtools bam2bm</a><a class="headerlink" href="#bmtools-bam2bm" title="Permalink to this heading">¶</a></h2>
<p>Calculate DNA methylation level from alignment BAM file to bm binary file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools bam2bm -g genome.fa -b sample.sort.bam -m sample.methratio.bm
</pre></div>
</div>
<p>you can see more details with 'bmtools bam2bm -h'</p>
</section>
<section id="bmtools-mr2bm">
<h2><a class="toc-backref" href="#id4" role="doc-backlink">bmtools mr2bm</a><a class="headerlink" href="#bmtools-mr2bm" title="Permalink to this heading">¶</a></h2>
<p>Convert methratio txt file to bm binary file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools mr2bm -g genome.fa.size -m mutant.methratio.txt --outbm mutant.methratio.bm
</pre></div>
</div>
<p>obtained bm format methylation results, you can see more details with 'bmtools mr2bm -h'</p>
</section>
<section id="bmtools-view">
<h2><a class="toc-backref" href="#id5" role="doc-backlink">bmtools view</a><a class="headerlink" href="#bmtools-view" title="Permalink to this heading">¶</a></h2>
<p>You can view and process bm file with bmtools:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools view -i mutant.methratio.bm <span class="p">|</span> head
  <span class="c1">#Chr1     34      35      0.600000        5       -       CHG</span>
  <span class="c1">#Chr1     80      81      0.333333        6       -       CHH</span>
  <span class="c1">#Chr1     116     117     1.000000        4       -       CG</span>
  <span class="c1">#Chr1     117     118     0.250000        4       -       CHG</span>
</pre></div>
</div>
<p>obtained text format methylation results</p>
</section>
<section id="bmtools-viewheader">
<h2><a class="toc-backref" href="#id6" role="doc-backlink">bmtools viewheader</a><a class="headerlink" href="#bmtools-viewheader" title="Permalink to this heading">¶</a></h2>
<p>You can view the format of bm file with bmtools:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools viewheader -i mutant.methratio.bm
  <span class="c1">#BM_END:    yes</span>
  <span class="c1">#BM_COVER:    yes</span>
  <span class="c1">#BM_CONTEXT:    yes</span>
  <span class="c1">#BM_STRAND:    yes</span>
  <span class="c1">#BM_ID:    no</span>
  <span class="c1">#Levels:     2</span>
  <span class="c1"># ...</span>
  <span class="c1">#Chromosome List</span>
  <span class="c1">#idx      Chrom   Length (bases)</span>
  <span class="c1">#0        Chr1    30427671</span>
  <span class="c1">#1        Chr2    19698289</span>
  <span class="c1"># ...</span>
</pre></div>
</div>
<p>obtained format of methylation results</p>
</section>
<section id="bmtools-overlap">
<h2><a class="toc-backref" href="#id7" role="doc-backlink">bmtools overlap</a><a class="headerlink" href="#bmtools-overlap" title="Permalink to this heading">¶</a></h2>
<p>Overlap cytosine site with more than two bm files:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools overlap -i sample1.methratio.bm -i2 sample2.methratio.bm
  <span class="c1">## chromsome pos context strand methy-sample1 coverage-sample1 methy-sample2 coverage-sample2</span>
  <span class="c1">#Chr1     34      CHG     -       0.600000        5       0.600000        5</span>
  <span class="c1">#Chr1     80      CHH     -       0.333333        6       0.333333        6</span>
  <span class="c1">#Chr1     116     CG      -       1.000000        4       1.000000        4</span>
  <span class="c1">#Chr1     117     CHG     -       0.250000        4       0.250000        4</span>
  <span class="c1">#Chr1     125     CHG     -       1.000000        4       1.000000        4</span>
</pre></div>
</div>
<p>Or just with --bmfiles:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools overlap --bmfiles sample1.methratio.bm sample2.methratio.bm
</pre></div>
</div>
</section>
<section id="bmtools-regionstats">
<h2><a class="toc-backref" href="#id8" role="doc-backlink">bmtools regionstats</a><a class="headerlink" href="#bmtools-regionstats" title="Permalink to this heading">¶</a></h2>
<p>Calculate DNA methylation level of chromosome region, genes, or TEs:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools regionstats -i sample1.methratio.bm --gtf gene.gtf -o gene.meth.txt --printcoverage <span class="m">1</span>
  <span class="c1">## chromosome pos strand meth coverage geneid</span>
  <span class="c1">#Chr1     4396348 -       6       567     AT1G12920</span>
  <span class="c1">#Chr1     4396348 -       12      1552    AT1G12920</span>
  <span class="c1">#Chr1     4398375 -       114     3381    AT1G12930</span>
</pre></div>
</div>
<p>or only print methylation level without coverage:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools regionstats -i sample1.methratio.bm --gtf gene.gtf -o gene.meth.txt --printcoverage <span class="m">0</span>
  <span class="c1">## chromosome pos methy-level geneid</span>
  <span class="c1">#Chr1     1618602 -       0.009665        AT1G05490</span>
  <span class="c1">#Chr1     1618602 -       0.014290        AT1G05490</span>
  <span class="c1">#Chr1     1624955 +       0.048446        AT1G05500</span>
  <span class="c1">#Chr1     1624955 +       0.213080        AT1G05500</span>
</pre></div>
</div>
<p>Or with bed file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools regionstats -i sample1.methratio.bm --bed gene.bed -o gene.meth.txt
</pre></div>
</div>
<p>Or just calculate DNA methylation level of same regions:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools regionstats -i sample1.methratio.bm -r chr1:1-2900<span class="p">;</span>chr2:1-200,+ <span class="se">\</span>
  -o gene.meth.txt
</pre></div>
</div>
<p>Please see 'bmtools regionstats' for more details.</p>
</section>
<section id="bmtools-bodystats">
<h2><a class="toc-backref" href="#id9" role="doc-backlink">bmtools bodystats</a><a class="headerlink" href="#bmtools-bodystats" title="Permalink to this heading">¶</a></h2>
<p>Calculate DNA methylation level of gene body, upstream and downstream:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools bodystats -i sample1.methratio.bm --gtf gene.gtf -o gene.meth.txt
</pre></div>
</div>
<p>Or with bed file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools bodystats -i sample1.methratio.bm --bed gene.bed -o gene.meth.txt
</pre></div>
</div>
<p>Or just calculate DNA methylation level of same regions:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools bodystats -i sample1.methratio.bm -r chr1:1-2900<span class="p">;</span>chr2:1-200,+ <span class="se">\</span>
  -o gene.meth.txt
</pre></div>
</div>
<p>Please see 'bmtools bodystats' for more details.</p>
</section>
<section id="bmtools-profile">
<h2><a class="toc-backref" href="#id10" role="doc-backlink">bmtools profile</a><a class="headerlink" href="#bmtools-profile" title="Permalink to this heading">¶</a></h2>
<p>Calculate DNA methylation profile matrix and avarage matrix across gene body, upstream and downstream:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools profile -i sample1.methratio.bm --gtf gene.gtf -o gene.profile <span class="se">\</span>
  --regionextend <span class="m">2000</span> --bodyX <span class="m">1</span> --matrixX <span class="m">5</span> --profilemode <span class="m">0</span>
</pre></div>
</div>
<p>Or with bed file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools profile -i sample1.methratio.bm --bed gene.bed -o gene.profile <span class="se">\</span>
  --regionextend <span class="m">2000</span> --bodyX <span class="m">1</span> --matrixX <span class="m">5</span> --profilemode <span class="m">0</span>
</pre></div>
</div>
<p>Please see 'bmtools profile' for more details.</p>
</section>
<section id="bmtools-chromstats">
<h2><a class="toc-backref" href="#id11" role="doc-backlink">bmtools chromstats</a><a class="headerlink" href="#bmtools-chromstats" title="Permalink to this heading">¶</a></h2>
<p>Calculate DNA methylation level across chromosome:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools chromstats -i sample1.methratio.bm -o chromosome.meth.txt <span class="se">\</span>
  --chromstep <span class="m">100000</span> --stepmove <span class="m">50000</span> --fstrand <span class="m">3</span> --context <span class="m">4</span>
</pre></div>
</div>
<p>Please see 'bmtools chromstats' for more details.</p>
</section>
<section id="bmtools-addzm">
<h2><a class="toc-backref" href="#id12" role="doc-backlink">bmtools addzm</a><a class="headerlink" href="#bmtools-addzm" title="Permalink to this heading">¶</a></h2>
<p>Add or change zoom levels in BM file, needed for IGV browser:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools addzm -i sample.mr.bm -o sample.mr.zm5.bm --zl <span class="m">5</span>
</pre></div>
</div>
<p>Please see 'bmtools addzm' for more details.</p>
</section>
<section id="bmtools-stats">
<h2><a class="toc-backref" href="#id13" role="doc-backlink">bmtools stats</a><a class="headerlink" href="#bmtools-stats" title="Permalink to this heading">¶</a></h2>
<p>Calculate DNA methylation data coverage and DNA methylation level category:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ bmtools stats -i sample1.methratio.bm -o chromosome.cover
</pre></div>
</div>
<p>Please see 'bmtools stats' for more details.</p>
<div class="admonition tip">
<p class="admonition-title">Tip</p>
<p>For feature requests or bug reports please open an issue <a class="reference external" href="http://github.com/ZhouQiangwei/BMtools">on github</a>.</p>
</div>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
    
        <div id="show_right_sidebar">
            <p><a class="toggle_right_sidebar" href="#"><span class="icon"><</span><span>Page contents<span></a></p>
        </div>

        <div id="right_sidebar">
            <p><a class="toggle_right_sidebar" href="#"><span class="icon">></span><span>Page contents:<span></a></p>
            <div class="page_toc">
                <ul>
<li><a class="reference internal" href="#">BMtools</a><ul>
<li><a class="reference internal" href="#bmtools-main-modules">BMtools main modules</a></li>
<li><a class="reference internal" href="#main-functions">Main functions</a></li>
<li><a class="reference internal" href="#bmtools-bam2bm">bmtools bam2bm</a></li>
<li><a class="reference internal" href="#bmtools-mr2bm">bmtools mr2bm</a></li>
<li><a class="reference internal" href="#bmtools-view">bmtools view</a></li>
<li><a class="reference internal" href="#bmtools-viewheader">bmtools viewheader</a></li>
<li><a class="reference internal" href="#bmtools-overlap">bmtools overlap</a></li>
<li><a class="reference internal" href="#bmtools-regionstats">bmtools regionstats</a></li>
<li><a class="reference internal" href="#bmtools-bodystats">bmtools bodystats</a></li>
<li><a class="reference internal" href="#bmtools-profile">bmtools profile</a></li>
<li><a class="reference internal" href="#bmtools-chromstats">bmtools chromstats</a></li>
<li><a class="reference internal" href="#bmtools-addzm">bmtools addzm</a></li>
<li><a class="reference internal" href="#bmtools-stats">bmtools stats</a></li>
</ul>
</li>
</ul>

            </div>
        </div>
    

      <div class="clearer"></div>
    </div>
    <div class="button_nav_wrapper">
        <div class="button_nav">
            <div class="left">
                
                <a href="../index.html">
                    <span class="icon"><</span><span>The Basic DNA Methylation (BM) format and BMtools</span></a>
                
            </div>

            <div class="right">
                
                    <a href="bam2bm.html"><span>Bam2bm</span><span class="icon">></span></a>
                
            </div>
        </div>
    </div>


    <div class="footer" role="contentinfo">
        &#169; Copyright 2022, momocoding.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 5.2.0.post0.
    </div>

<p id="theme_credit">Styled using the <a href="https://github.com/piccolo-orm/piccolo_theme">Piccolo Theme</a></p>
  </body>
</html>